src/ripple/app/paths/impl/StrandFlow.h

```diff
@@ -144,14 +144,21 @@ flow (
                         *sb, *afView, ofrsToRm, EitherAmount (*maxIn));
                     limitStepOut = r.second;
 
-                    if (strand[i]->isZero (r.second) ||
-                        get<TInAmt> (r.first) != *maxIn)
+                    if (strand[i]->isZero(r.second))
+                    {
+                        JLOG(j.trace()) << "First step found dry";
+                        return {tecPATH_DRY, std::move(ofrsToRm)};
+                    }
+                    if (get<TInAmt> (r.first) != *maxIn)
                     {
                         // Something is very wrong
                         // throwing out the sandbox can only increase liquidity
                         // yet the limiting is still limiting
-                        JLOG (j.fatal()) << "Re-executed limiting step failed";
-                        assert (0);
+                        JLOG(j.fatal())
+                            << "Re-executed limiting step failed. r.first: "
+                            << to_string(get<TInAmt>(r.first))
+                            << " maxIn: " << to_string(*maxIn);
+                        assert(0);
```

Comment:
I would like to see us develop a way to bring a run-time issue to our attention without terminating the process unless we believe that the process is corrupted to the point that it would be unsafe for it to continue.  I'm not sure what such a process would look like, but improved access to the logs of validators we control would be a step in the right direction.

Reply:
@HowardHinnant You are correct that the purpose of the assert is to bring the issue to our attention, and the process is safe to continue (if it were not I'd throw a logic error). 

I'm coming from the assumption that `asserts` are not used in production (validators that use our package are built with `NDEBUG` so the asserts won't fire). So long as that is true, the `asserts` are doing what I want. 

Are you worried that someone is going to turn off `NDEBUG` on the production machines, or do you object to bringing down the process even on dogfood machines? I'll try to think of a different way to do this, but I do think it's safe to merge with the `assert` as-is if we don't think of another way.

Reply:
I'm concerned that the program with asserts enabled behaves differently than without.  It would be better if we could depend on the asserts that might be downstream of returning `telFAILED_PROCESSING`.

Reply:
One solution would be to remove the asserts and set a breakpoint there on dogfood machines. The downside to that is I'd want to make sure that everyone on the team ran their dogfood machines that way. I still want to keep the asserts, but I'd be OK with this solution too.

Reply:
I'm not loosing sleep over this.  I've tested this patch and everything looks ok to me.  But I did want to start this conversation.  E.g. what if we want to debug the case that `telFAILED_PROCESSING` get returned and something unexpected happens afterward.  Or maybe we just want to assure that it is safe to keep the process up afterwards.

Reply:
I DM'd @HowardHinnant and I do see where he's coming from. However, we agreed that it's OK to keep the asserts for now (and we'll look for a better solution for this class of problems - maybe an external tool).